x/auth/middleware/tips.go

```diff
@@ -26,57 +24,57 @@ func NewTipMiddleware(bankKeeper types.BankKeeper) tx.Middleware {
 var _ tx.Handler = tipsTxHandler{}
 
 // CheckTx implements tx.Handler.CheckTx.
-func (txh tipsTxHandler) CheckTx(ctx context.Context, sdkTx sdk.Tx, req abci.RequestCheckTx) (abci.ResponseCheckTx, error) {
-	res, err := txh.next.CheckTx(ctx, sdkTx, req)
+func (txh tipsTxHandler) CheckTx(ctx context.Context, req tx.Request, checkTx tx.RequestCheckTx) (tx.Response, tx.ResponseCheckTx, error) {
+	res, resCheckTx, err := txh.next.CheckTx(ctx, req, checkTx)
 	if err != nil {
-		return abci.ResponseCheckTx{}, err
+		return tx.Response{}, tx.ResponseCheckTx{}, err
 	}
 
-	tipTx, ok := sdkTx.(tx.TipTx)
+	tipTx, ok := req.Tx.(tx.TipTx)
 	if !ok || tipTx.GetTip() == nil {
-		return res, err
+		return res, tx.ResponseCheckTx{}, err
 	}
 
 	if err := txh.transferTip(ctx, tipTx); err != nil {
-		return abci.ResponseCheckTx{}, err
+		return tx.Response{}, tx.ResponseCheckTx{}, err
 	}
 
```

Comment:
hmm, i'm noticing that since CheckTx has a different return signature, we can't factorize as much as I thought